### ISSUE TITLE
function to get Justfile location path

### ISSUE THREAD

**Issue by camerondavison** [casey/just#387]
I created a super simple `Justfile`
```
info:
	touch file.empty
```
then created a nested project.
```
mkdir -p a/b/c
```
then ran 
```
just --justfile Justfile --working-directory .
just --justfile Justfile --working-directory a
just --justfile Justfile --working-directory a/b
just --justfile Justfile --working-directory a/b/c
```
and got
```
$ tree
.
├── Justfile
├── a
│   ├── b
│   │   ├── c
│   │   │   └── file.empty
│   │   └── file.empty
│   └── file.empty
└── file.empty
```
I am going to assume that this is expected, but given https://github.com/casey/just/issues/312 and the `invocation_directory` function it seems like there is not a great way to do something like
```
ROOT_DIR:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
```
that you can do in make to allow paths that are absolute, but relative to the Justfile.

I would like to make paths absolute so that I do not depend on someone calling `--justfile` with the correct `--working-directory`

**Comment by casey** [comment]
This should be easy to support. If you wouldn't mind elaborating on your use case, that would be great, to make sure the feature actually does what you want to do.

Functions are defined in `functions.rs`. They get passed a `FunctionContext`, which contains the information they need to function. (Haha.) It would be pretty easy to add a function that returns the directory the `justfile` was found in.

**Comment by camerondavison** [comment]
the use case is I have something like
```justfile
clean:
     rm -rf build
```
I would prefer to write
```justfile
clean:
     rm -rf "{{source_dir()}}/build"
```
or even better
```justfile
project_dir = source_dir()
build_dir = project_dir + "/build"

clean:
     rm -rf "{{build_dir}}"
```

**Comment by casey** [comment]
That makes sense.

Currently, just requires `--working-directory` when `--justfile` is set. This should be changed so that you can pass `--justfile` without `--working-directory`, issue #340. There's actually an open PR, #355, that just needs an integration test before it lands.

Why is the justfile being called with `--justfile` and invoked by letting `just` find the justfile itself and setting the working directory? Would not requiring `--working-directory` with `--justfile` be a good fix?

**Comment by camerondavison** [comment]
My thinking is that people "could" call it that way. And I would never want to accidentally delete something in the wrong working directory. Which is why I always like to make sure the path is absolute. This can be done by making it relative to the justfile but then knowing the absolute location of the justfile.

**Comment by casey** [comment]
Gotcha, that makes sense. I personally don't worry about that, but being extra careful for some commands (`rm -rf bin`, etc) is reasonable.

This wouldn't be too hard to implement, if someone's up to the task. The location of the justfile needs to be added to `FunctionContext`, and a new function created. Maybe `justfile_directory()?`

**Comment by dionjwa** [comment]
Another use case is where you have multiple justfiles (a shared set of tasks, and a specific set of tasks, maybe solved by imports).

One justfile is called `ci` and is executable. However, it can't figure out itself that it's path is `./ci` so `--list` is broken, and anything that assumes that the path is `justfile`.

**Comment by casey** [comment]
I just landed #569, which includes both `justfile()` and `justfile_directory()`. Let me know if you run into any issues with them!
